build/gh-pages-app.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const cpy = require('cpy');
+const ghPages = require('gh-pages');
+const glob = require('glob');
+const lighthousePackage = require('../package.json');
+const rimraf = require('rimraf');
+const terser = require('terser');
+
+const ghPagesDistDir = `${__dirname}/../dist/gh-pages`;
+
+const license = `/*
+* @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+* or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/`;
+
+/**
+ * @typedef BuildOptions
+ * @property {string} name
+ * @property {string} appDir
+ * @property {string} htmlPath
+ * @property {Array<{path: string} | string>} stylesheets
+ * @property {Array<{path: string} | string>} javascripts
+ * @property {string[]} assetPaths
```

Comment:
it makes more sense to me to make all four of these `Array<{path: string} | string>` if some of them are going to go this way, but it's not a huge deal

Reply:
Can't do it for assetPaths without having some way to interpret a literal string as an asset.  `{path: string}[]` might work?

Concating an array of html files doesn't make sense, but perhaps `{path: string} | string` could be allowed?

Reply:
Oh right, cause assets are just copied over, nothing new is created.

Yeah, maybe having them be be `{path: string}` helps self document by making them consistent with the others, but it also makes sense how you have it here.